src/main/java/frc/robot/constants/Constants.java

```diff
@@ -19,8 +19,18 @@ public final class Constants {
   public static final double UP_ADJUST = 0.5;
   public static final double DOWN_ADJUST = 0.25;
 
-  public static final Translation2d BLUE_ORBIT_POSE = new Translation2d(0.5, 5.5);
-  public static final Translation2d RED_ORBIT_POSE = new Translation2d(16, 5.5);
+  // TODO should revise and precisely measure these values, they are estimates
+  public static final Translation2d BLUE_SPEAKER_POSE = new Translation2d(0, 5.5);
+  public static final Translation2d RED_SPEAKER_POSE = new Translation2d(16.5, 5.5);
+  public static final double SPEAKER_HEIGHT = 2.1;
+  public static final Supplier<Translation2d> SPEAKER_TRANSLATION =
+      () ->
+          DriverStation.getAlliance().isPresent()
+              ? (DriverStation.getAlliance().get() == Alliance.Red
+                  ? RED_SPEAKER_POSE
+                  : BLUE_SPEAKER_POSE)
+              : BLUE_SPEAKER_POSE;
```

Comment:
TODO: don't poll the alliance every time. Only poll in disabled periodic

Reply:
Add a todo / make a ticket. Fine for this PR

Reply:
Done without ticket